src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs

```diff
@@ -172,10 +241,30 @@ private void AssertNotVisitingAfterGap()
         {
             if (_gapStart != null)
             {
-                throw new InvalidOperationException($"Not expecting to visit block at {_currentLevelNumber} because the gap has already been identified.");
+                throw new InvalidOperationException(
+                    $"Not expecting to visit block at {_currentLevelNumber} because the gap has already been identified.");
+            }
+        }
+
+        private bool CanSuggestBlocks(Block block)
+        {
+            _firstBlockVisited = false;
+            if (block?.ParentHash != null)
+            {
+                Block parentBlock = _blockTree.FindBlock(block.ParentHash,
```

Comment:
parentBlock.FindParent blockw ould be much more efficient here?

Reply:
public static BlockHeader? FindParentHeader(this IBlockFinder finder, BlockHeader header, BlockTreeLookupOptions options)